astropy/coordinates/baseframe.py

```diff
@@ -945,6 +945,32 @@ def is_frame_attr_default(self, attrnm):
         """
         return attrnm in self._attr_names_with_defaults
 
+    def is_equivalent_frame(self, other):
+        """
+        Checks if this object is the same frame as the ``other`` object.
+
+        To be the same frame, two objects must be the same frame class and have
+        the same frame attributes.  Note that it does *not* matter what, if any,
+        data either object has.
+
+        Parameters
+        ----------
+        other : BaseCoordinateFrame
+            the other frame to check
+
+        Returns
+        -------
+        isequiv : bool
+            True if the frames are the same, False if not.
+        """
+        if self.__class__ == other.__class__:
```

Comment:
If you test for `self.name == other.name` then it seems this would also allow `other` being a `SkyCoord`.  It will only check the available frame attributes in `self`, so if there are other attrs defined in the `SkyCoord` they would be "don't care".  That *seems* OK to me but not sure.